DHFPROD-2942:Backend task for custom ingestion in QS

srinathgit opened the pull request:
This PR addresses DHFPROD-2942, DHFPROD-2950

akshaysonvane commented on web/src/main/java/com/marklogic/hub/web/service/FlowManagerService.java:
```diff
@@ -217,48 +214,17 @@ public StepModel createStep(String flowName, Integer stepOrder, String stepId, S
+        boolean invalidStepType = true;
+        for (StepDefinition.StepDefinitionType stepType : StepDefinition.StepDefinitionType.values()) {
+            if (stepType.equals(step.getStepDefinitionType())) {
+                invalidStepType = false;
```
Just a nitpick, we can get rid of `invalidStepType` flag and throw the exception from here if we negate the if condition.

akshaysonvane commented on web/src/main/java/com/marklogic/hub/web/service/FlowManagerService.java:
```diff
@@ -295,6 +261,36 @@ public StepModel createStep(String flowName, Integer stepOrder, String stepId, S
+        if (stepDefinitionManagerService.getStepDefinition(step.getStepDefinitionName(), step.getStepDefinitionType()) != null) {
+            String stepType = step.getStepDefinitionType().toString().toLowerCase();
+            if(step.getStepDefinitionName().equalsIgnoreCase("default-" + stepType)) {
+                StepDefinition defaultStepDefinition = getDefaultStepDefinitionFromResources("hub-internal-artifacts/step-definitions/" + stepType + "/marklogic/default-"+ stepType +".step.json", step.getStepDefinitionType());
```
+1 for consolidating code.

aebadirad commented:
Let's make sure to work in https://github.com/marklogic/marklogic-data-hub/tree/DHFPROD-2735-custom-scaffold changes here.

srinathgit commented:
I have made the following changes 
1. Removed 'modulePath' from Step and added 'batchSize' and 'threadCount' to StepModel (since Nikhil will be adding them in UI for steps)
2. Used EnumUtils to check if a step is valid
3. CherryPicked Alex's latest commit from https://github.com/marklogic/marklogic-data-hub/tree/DHFPROD-2735-custom-scaffold

srinathgit commented:
@aebadirad ,
After incorporating the last commit from the branch https://github.com/marklogic/marklogic-data-hub/tree/DHFPROD-2735-custom-scaffold, mapping step fails with stackoverflow error .

srinathgit commented:
@xnikhil08 ,
Don't merge this to the feature branch yet

srinathgit commented:
Bringing in the changes to feature branch
